### PR TITLE
Better validation screens for the importer

### PR DESCRIPTION
**Pull request by eanders**

### CODE REVIEW

File: drivers/hmis_csv_twenty_twenty/app/controllers/hmis_csv_twenty_twenty/importer_errors_controller.rb [hunk 1]
```diff
@@ -11,7 +11,7 @@ def show
     importer_log = HmisCsvTwentyTwenty::Importer::ImporterLog.find(params[:id].to_i)
     @import = GrdaWarehouse::ImportLog.find_by(importer_log_id: importer_log.id)
 
-    @filename = HmisCsvTwentyTwenty::Importer::Importer.importable_files_map.keys.detect { |v| v == params[:file] }
+    @filename = HmisCsvTwentyTwenty::Importer::Importer.importable_files_map.keys.detect { |v| v == "#{params[:file]}.csv" }
```
Comment (awisema) [inline drivers/hmis_csv_twenty_twenty/app/controllers/hmis_csv_twenty_twenty/importer_errors_controller.rb]: Use `ValidationFiltering` here too?

File: drivers/hmis_csv_twenty_twenty/app/models/hmis_csv_validation/inclusion_in_set.rb [hunk 2]
```diff
@@ -13,7 +13,18 @@ def self.check_validity!(item, column, valid_options: nil)
...
       source_id: item.source_id,
       source_type: item.source_type,
-      status: "Expected #{value} to be included in #{valid_options} for #{column}",
+      status: "Expected #{value} to be included in #{short_list(valid_options)} for #{column}",
+      validated_column: column,
     )
   end
+
+  def self.short_list(options)
+    return options if options.length < 10
```
Comment (awisema) [inline drivers/hmis_csv_twenty_twenty/app/models/hmis_csv_validation/inclusion_in_set.rb:22]: These bounds don't match?

File: drivers/hmis_csv_twenty_twenty/app/views/hmis_csv_twenty_twenty/error_shared/_table.haml [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+      %thead
+        %tr
+          %th Error
+          %th HUD ID
+      %tbody
+        - errors.each do |error|
+          %tr
+            %td= error.details
+            - source = error.source
+            %td= source.send(source.class.hud_key) if source.present?
```
Comment (awisema) [inline drivers/hmis_csv_twenty_twenty/app/views/hmis_csv_twenty_twenty/error_shared/_table.haml]: An idle thought -- what are the trade-offs of `send` vs `public_send`?

**Review by awisema — CHANGES_REQUESTED** [review]

File: drivers/hmis_csv_twenty_twenty/app/controllers/hmis_csv_twenty_twenty/importer_errors_controller.rb (hunk 1 above)
Comment (eanders) [reply drivers/hmis_csv_twenty_twenty/app/controllers/hmis_csv_twenty_twenty/importer_errors_controller.rb]: Yes

File: drivers/hmis_csv_twenty_twenty/app/views/hmis_csv_twenty_twenty/error_shared/_table.haml (hunk 3 above)
Comment (eanders) [reply drivers/hmis_csv_twenty_twenty/app/views/hmis_csv_twenty_twenty/error_shared/_table.haml]: public_send with throw an error if the method is private, so generally, it's a safer approach.

**Pull request merged**